src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -107,26 +152,32 @@ static string FormatCustomAttribute (CustomAttribute ca)
 			var objectType = BCL.FindPredefinedType ("System", "Object", _context);
 			if (objectType == null)
 				return null;
+			var objectArrayType = new ArrayType (objectType);
+			if (objectArrayType == null)
+				return null;
 
 			//
 			// Generates metadata information for internal type
 			//
 			// public sealed class RemoveAttributeInstancesAttribute : Attribute
 			// {
 			//		public RemoveAttributeInstancesAttribute () {}
-			//		public RemoveAttributeInstancesAttribute (object value1) {}
+			//		public RemoveAttributeInstancesAttribute (params object[] values) {}
 			// }
 			//
-			var td = new TypeDefinition ("", "RemoveAttributeInstancesAttribute", TypeAttributes.Public);
-			td.BaseType = attributeType;
-
 			const MethodAttributes ctorAttributes = MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.SpecialName | MethodAttributes.RTSpecialName | MethodAttributes.Final;
-			var ctor = new MethodDefinition (".ctor", ctorAttributes, voidType);
-			td.Methods.Add (ctor);
 
-			ctor = new MethodDefinition (".ctor", ctorAttributes, voidType);
-			ctor.Parameters.Add (new ParameterDefinition (objectType));
-			td.Methods.Add (ctor);
+			if (td == null) {
+				td = new TypeDefinition ("", nameof (RemoveAttributeInstancesAttribute), TypeAttributes.Public);
+
+				td.BaseType = attributeType;
+			}
+
+			var ctorN = new MethodDefinition (".ctor", ctorAttributes, voidType);
+			var param = new ParameterDefinition (objectArrayType);
+			param.CustomAttributes.Add (ParamArrayCustomAttribute);
```

Comment:
Nit: technically there's no reason to add this attribute here - linker doesn't care, it will consume the object array directly. The attribute is really only useful for compilers when somebody tries to call the method, the compiler knows that it should pass additional parameters as an array. At runtime it's mostly ignored.